src/bin/pg_autoctl/pgctl.c

```diff
@@ -1172,6 +1172,82 @@ prepare_guc_settings_from_pgsetup(const char *configFilePath,
 }
 
 
+bool
+ensure_empty_tablespace_dirs(const char *pgdata)
+{
+	char *dirName = "pg_tblspc";
+	struct dirent *tblspcDirEntry = NULL;
+
+	char pgTblspcFullPath[MAXPGPATH] = { 0 };
+	sformat(pgTblspcFullPath, MAXPGPATH, "%s/%s", pgdata, dirName);
+
+	if (!directory_exists(pgTblspcFullPath))
+	{
+		log_debug("Postgres dir pg_tblspc does not exist at \"%s\"",
+				  pgTblspcFullPath);
+		return true;
+	}
+
+
+	/* open and scan through the Postgres tablespace directory */
+	DIR *tblspcDir = opendir(pgTblspcFullPath);
+
+	if (tblspcDir == NULL)
+	{
+		log_error("Failed to open Postgres tablespace directory \"%s\" at \"%s\"",
+				  dirName, pgdata);
+		return false;
+	}
+
+	while ((tblspcDirEntry = readdir(tblspcDir)) != NULL)
+	{
+		char tblspcDataDirPath[MAXPGPATH] = { 0 };
+
+		/* skip non-symlinks, as all tablespace dirs are symlinks */
+		if (tblspcDirEntry->d_type == DT_UNKNOWN)
+		{
+			struct stat structStat;
+			char pgTblspcEntryFullPath[MAXPGPATH] = { 0 };
+			sformat(pgTblspcEntryFullPath, MAXPGPATH, "%s/%s", pgTblspcFullPath,
+					tblspcDirEntry->d_name);
+			if (lstat(pgTblspcEntryFullPath, &structStat) != 0)
+			{
+				log_error("Failed to get file information for \"%s/%s\": %m",
+						  pgTblspcFullPath,
+						  tblspcDirEntry->d_name);
+				return false;
+			}
+			if (!S_ISLNK(structStat.st_mode))
+			{
+				continue;
```

Comment:
We might want to log_debug about those non-symlink entries found in pg_tblspc?

Reply:
We'd be logging about `.` and `..` every time, but it's debug log so could be ok?

Reply:
We might want to skip those too, that's a good comment, thanks!